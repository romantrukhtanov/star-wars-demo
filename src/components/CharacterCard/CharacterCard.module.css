.characterCard,
.actionArea {
  height: 100%;
}

.name {
  font-size: 16px;
}

.gender {
  font-size: 14px;
}

@media (max-width: 767.98px) {
  .name {
    font-size: 14px;
  }

  .text {
    font-size: 12px;
  }
}
