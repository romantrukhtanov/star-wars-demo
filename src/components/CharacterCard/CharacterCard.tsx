import { Card, CardActionArea, CardContent, CardMedia, Typography } from '@mui/material';
import { Link } from 'react-router-dom';

import type * as M from '@/core/api/model';

import styles from './CharacterCard.module.css';

type Props = {
  character: Pick<M.Character, 'url' | 'name' | 'gender' | 'id' | 'imageUrl'>;
};

export function CharacterCard({ character }: Props) {
  return (
    <Card className={styles.characterCard}>
      <CardActionArea className={styles.actionArea} component={Link} to={character.url} state={character}>
        <CardMedia className={styles.media} component="img" image={character.imageUrl} alt={character.name} />
        <CardContent>
          <Typography className={styles.name} color="text.primary" variant="h3">
            {character.name}
          </Typography>
          <Typography className={styles.gender} color="text.secondary">
            {character.gender}
          </Typography>
        </CardContent>
      </CardActionArea>
    </Card>
  );
}
