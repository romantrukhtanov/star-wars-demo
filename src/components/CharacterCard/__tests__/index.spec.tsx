import { describe, expect, it } from 'vitest';

import { render } from '@/tests/test-utils';

import { CharacterCard } from '../CharacterCard';

const mockData = {
  name: 'Leia Organa',
  gender: 'female',
  url: 'https://swapi.dev/api/people/5/',
  id: '5',
  imageUrl: '/images/people/5.jpg',
};

describe('CharacterCard', () => {
  it('render component', () => {
    const { baseElement } = render(<CharacterCard character={mockData} />);
    expect(baseElement).toMatchSnapshot();
  });
});
