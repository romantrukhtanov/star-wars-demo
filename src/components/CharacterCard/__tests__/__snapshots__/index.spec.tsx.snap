// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CharacterCard > render component 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root _characterCard_c7dba3 css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <a
        class="MuiButtonBase-root MuiCardActionArea-root _actionArea_c7dba3 css-1oitmrd-MuiButtonBase-root-MuiCardActionArea-root"
        href="https://swapi.dev/api/people/5/"
        tabindex="0"
      >
        <img
          alt="Leia Organa"
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img _media_c7dba3 css-o69gx8-MuiCardMedia-root"
          src="/images/people/5.jpg"
        />
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3 _name_c7dba3 css-krdrhq-MuiTypography-root"
          >
            Leia Organa
          </h3>
          <p
            class="MuiTypography-root MuiTypography-body1 _gender_c7dba3 css-1pnmrwp-MuiTypography-root"
          >
            female
          </p>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
    </div>
  </div>
</body>
`;
